src/compiler/checker.ts

```diff
@@ -3112,6 +3112,9 @@ namespace ts {
                         return "(Anonymous function)";
                 }
             }
+            if (hasProperty(symbol, "syntheticName")) {
+                return (symbol as TransientSymbol).syntheticName;

```

Comment:
can you use `symbol.syntheticOrigin.name`? (or escapedName, not sure which). Even if it doesn't exist, creating a synthetic symbol for syntheticOrigin might make more sense than creating a standalone name.

Reply:
No - the symbol will have no `syntheticOrigin`, as it has no source type to draw a named declaration from.